r/tests/testthat/test-dplyr-string-functions.R

```diff
@@ -18,6 +18,27 @@
 library(dplyr)
 library(stringr)
 
+test_that("sub and gsub with ignore.case = FALSE and fixed = TRUE", {
+  df <- tibble(x = c("Foo", "bar"))
+
+  expect_dplyr_equal(
+    input %>%
+      transmute(x = sub("Foo", "baz", x, fixed = TRUE)) %>%
+      collect(),
+    df
+  )
+  expect_dplyr_equal(
+    input %>%
+      transmute(x = gub("o", "u", x, fixed = TRUE)) %>%
+      collect(),
+    df
+  )
+
+})
+
+# many of the remainder of these tests require RE2
+skip_if_not_available("re2")
+
```

Comment:
In ARROW-11338, I forgot to add a check for RE2 before running tests that require it. Here I add that check, and also add a couple of simple tests before it that don't use RE2.